app/controllers/application_controller.rb

```diff
@@ -13,6 +13,37 @@
 # Likewise, all the methods added will be available for all controllers.
 
 class ApplicationController < ActionController::Base
+ 
+  before_action :set_locale
+
+  def set_locale    
+    I18n.locale = cookies[:locale] || detect_locale
+  end
+
+  def detect_locale
+    
+    binding.pry
+    
+    # Parse the Accept-Language header and split it into an array of language codes
+    languages = request.headers['Accept-Language']&.split(',')
+
+    language_locale_map = {
+      'en' => :en,
+      'fr' => :fr
+    }
+
+    languages.each do |language|
+      # Extract the language code from the language string
+      language_code = language.split(';').first
+
+      # If the language code is in our map, return the corresponding locale
+      return language_locale_map[language_code] if language_locale_map[language_code]
```

Comment:
can be replaced with `language_code.to_sym`, no ?

Reply:
I actually wanted to verify whether the `language_code `is included in the `supported_languages`. If it's not supported, then we will set the platform language to the default one.

 I will refactor the code to make this more clearer.